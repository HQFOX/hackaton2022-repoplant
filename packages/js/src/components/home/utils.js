import {
  Level1,
  Level2Average,
  Level3Bad,
  Level4,
  Level5
} from "@hv/uikit-react-icons";

export const getStatus = statusNumber => {
  switch (statusNumber) {
    case 1:
      return { Icon: Level1, sema: "sema10" };
    case 2:
      return { Icon: Level2Average, sema: "sema11" };
    case 3:
      return { Icon: Level3Bad, sema: "sema12" };
    case 4:
      return { Icon: Level4, sema: "sema13" };
    case 5:
      return { Icon: Level5, sema: "sema14" };
    default:
      return { Icon: null, sema: "sema1" };
  }
};
