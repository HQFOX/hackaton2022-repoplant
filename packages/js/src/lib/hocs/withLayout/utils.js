export const findById = (data = [], id) => {
  for (let i = 0; i < data.length; i += 1) {
    const el = data[i];
    if (el.id === id) return el;

    const child = findById(el.data, id);
    if (child) return child;
  }
  return null;
};

export const findLeaf = item => (item.data ? findLeaf(item.data[0]) : item);

export const findFirstChildByParentId = (data = [], id) => {
  const parent = findById(data, id);
  return (parent && parent.data && findLeaf(parent.data[0]).id) || id;
};

export const navigationDepth = arr =>
  Array.isArray(arr)
    ? 1 + Math.max(...arr.map(el => navigationDepth(el.data)))
    : 0;
