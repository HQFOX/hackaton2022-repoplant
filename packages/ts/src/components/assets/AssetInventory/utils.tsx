import React from "react";
import {
  AssetInventoryMetadata,
  AssetInventoryConfiguration,
  HvSemanticColorKeys
} from "@hv/uikit-react-core";
import {
  Delete,
  Preview,
  Upload,
  Level1,
  Level2Average,
  Level3Bad,
  Level4,
  Level5
} from "@hv/uikit-react-icons";

const metadata: AssetInventoryMetadata[] = [
  {
    id: "id1",
    accessor: "headerTitle",
    cellType: "alpha-numeric",
    searchable: true,
    sortable: true,
    sortableLabelAsc: "Title ascending",
    sortableLabelDesc: "Title descending"
  },
  {
    id: "id2",
    accessor: "semantic",
    cellType: "alpha-numeric"
  },
  {
    id: "id3",
    accessor: "probability",
    cellType: "numeric",
    searchable: true,
    sortable: true,
    sortableLabelAsc: "Probability ascending",
    sortableLabelDesc: "Probability descending"
  },
  {
    id: "id4",
    accessor: "timeHorizon",
    cellType: "numeric",
    sortable: true,
    sortableLabelAsc: "TimeHorizon ascending",
    sortableLabelDesc: "TimeHorizon descending"
  },
  {
    id: "id5",
    accessor: "event.schedule",
    cellType: "alpha-numeric",
    searchable: true
  },
  {
    id: "id6",
    accessor: "event.description",
    cellType: "alpha-numeric",
    searchable: true
  }
];

const viewConfiguration: AssetInventoryConfiguration = {
  breakpoints: { xs: 12, sm: 6, md: 4, lg: 3, xl: 3 },
  columnConfiguration: [
    { style: { width: 1, textAlign: "center" } },
    {
      title: "Event",
      style: { minWidth: "360px", textAlign: "start" }
    },
    {
      title: "Probability",
      style: { minWidth: "80px", textAlign: "end" }
    },
    {
      title: "Time horizon",
      style: { minWidth: "100px", textAlign: "end" }
    },
    {
      title: "Related Assets",
      style: {
        minWidth: "195px",
        paddingLeft: "30px",
        textAlign: "start"
      }
    }
  ]
};

const configuration = {
  metadata,
  viewConfiguration
};

const actions = [
  { id: "post", label: "Dismiss", disabled: false },
  {
    id: "get",
    label: "Preview",
    icon: <Preview color="atmo5" />,
    disabled: true
  },
  {
    id: "put",
    label: "Upload",
    icon: <Upload color="atmo5" />,
    disabled: true
  },
  { id: "delete", label: "Delete", icon: <Delete />, disabled: false }
];

type Status = {
  Icon: React.ComponentType<any> | null;
  sema: HvSemanticColorKeys;
};

const getStatus = (statusNumber: number): Status => {
  switch (statusNumber) {
    case 1:
      return { Icon: Level1, sema: "sema10" };
    case 2:
      return { Icon: Level2Average, sema: "sema11" };
    case 3:
      return { Icon: Level3Bad, sema: "sema12" };
    case 4:
      return { Icon: Level4, sema: "sema13" };
    case 5:
      return { Icon: Level5, sema: "sema14" };
    default:
      return { Icon: null, sema: "sema1" };
  }
};

export { configuration, actions, getStatus };
