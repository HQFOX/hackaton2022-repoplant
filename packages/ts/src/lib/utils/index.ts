export const getRandom = (max: number, min = 1) =>
  Math.floor(Math.random() * (max - min + 1)) + min;
