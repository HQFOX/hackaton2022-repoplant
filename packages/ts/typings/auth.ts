// Use enums for better autocompletion of action type names.
//
// We use the `@@context/ACTION_TYPE` convention,
// to follow the convention of Redux's `@@INIT` action.
export enum AuthActions {
  SET_AUTH = "@@auth/SET_AUTH"
}

// Declare state types with `readonly` modifier to get compile time immutability.
// https://github.com/piotrwitek/react-redux-typescript-guide#state-with-type-level-immutability
export interface AuthState {
  readonly isAuthed: boolean | null;
}

export interface AuthProps {
  login(credentials: AuthCredentials): void;
}

export interface AuthCredentials {
  username: string;
  password: string;
}
