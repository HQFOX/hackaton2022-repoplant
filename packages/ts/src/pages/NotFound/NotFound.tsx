import React from "react";
import HvTypography from "@hv/uikit-react-core/dist/Typography";
import HvGrid from "@hv/uikit-react-core/dist/Grid";
import { useTranslation } from "react-i18next";
import withLayout from "lib/hocs/withLayout";
import { NotFoundProps } from "./index";

const NotFound: React.FC<NotFoundProps> = ({ classes }: NotFoundProps) => {
  const { t } = useTranslation();

  return (
    <HvGrid container>
      <HvGrid item xl={12}>
        <HvTypography variant="3xlTitle" className={classes.title}>
          {t("pages.notFound.title")}
        </HvTypography>
      </HvGrid>
    </HvGrid>
  );
};

export default withLayout(NotFound);
