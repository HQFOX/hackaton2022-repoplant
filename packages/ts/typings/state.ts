import { RouterState } from "connected-react-router";
import { AuthState } from "./auth";
import { DataState } from "./data";
import { PagesState } from "./pages";

export type DeepPartial<T> = T extends Function
  ? T
  : T extends object
  ? { [P in keyof T]?: DeepPartial<T[P]> }
  : T;

// The top-level state object
export interface AppState {
  readonly auth: AuthState;
  readonly data: DataState;
  readonly pages: PagesState;
  readonly router: RouterState;
}

export type TestAppState = DeepPartial<AppState>;
