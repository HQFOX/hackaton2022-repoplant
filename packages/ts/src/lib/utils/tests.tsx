import React from "react";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { HvProvider } from "@hv/uikit-react-core";
import { TestAppState } from "typings/state";

export const createMockStore = (state: TestAppState): any => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore(state);
  return store;
};

export const withStoreProvider = <P extends {}>(
  Component: React.ComponentType<P>,
  state: TestAppState
): React.FC<P> => props => (
  <Provider store={createMockStore(state)}>
    <HvProvider>
      <MemoryRouter initialEntries={[{ pathname: "/", key: "testKey" }]}>
        <Component {...props} />
      </MemoryRouter>
    </HvProvider>
  </Provider>
);

export const withHvProvider = <P extends {}>(
  Component: React.ComponentType<P>
): React.FC<P> => props => (
  <HvProvider>
    <MemoryRouter initialEntries={[{ pathname: "/", key: "testKey" }]}>
      <Component {...props} />
    </MemoryRouter>
  </HvProvider>
);
