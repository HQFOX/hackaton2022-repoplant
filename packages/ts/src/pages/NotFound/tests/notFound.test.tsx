import React from "react";
import { mount } from "enzyme";
import { withStoreProvider } from "lib/utils/tests";
import { TestAppState } from "typings/state";
import NotFound from "..";

describe("<NotFound />", () => {
  let snapshot;

  beforeEach(() => {
    const WithStoreProvider = withStoreProvider(NotFound, {
      router: { location: { pathname: "/" } },
      auth: { isAuthed: false },
      pages: { data: [] },
      logout: jest.fn()
    } as TestAppState);
    snapshot = mount(<WithStoreProvider />);
  });

  test("it matches the snapshot", () => {
    expect(snapshot).toMatchSnapshot();
  });
});
