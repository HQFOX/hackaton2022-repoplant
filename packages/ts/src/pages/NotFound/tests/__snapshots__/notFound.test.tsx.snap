// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> it matches the snapshot 1`] = `undefined`;
